src/components/custom-field-input-date/format/format-date.js

```diff
@@ -0,0 +1,92 @@
+const months = ['january', 'february', 'march', 'april', 'may', 'june',
+  'july', 'august', 'september', 'october', 'november', 'december'];
+
+export const validFormats = {
+  'mm/dd/yyyy': {
+    matcher: /^\d{2}\/\d{2}\/\d{4}$/,
+    combine: dismantledDate => `${dismantledDate.month}/${dismantledDate.day}/${dismantledDate.year}`,
+    dismantle: (string) => {
+      const pieces = string.split('/');
+
+      return {
+        day: pieces[1],
+        month: pieces[0],
+        year: pieces[2],
+      };
+    },
+  },
+  'mm-dd-yyyy': {
+    matcher: /^\d{2}-\d{2}-\d{4}$/,
+    combine: dismantledDate => `${dismantledDate.month}-${dismantledDate.day}-${dismantledDate.year}`,
+    dismantle: (string) => {
+      const pieces = string.split('-');
+
+      return {
+        day: pieces[1],
+        month: pieces[0],
+        year: pieces[2],
+      };
+    },
+  },
+  'yyyy-mm-dd': {
+    matcher: /^\d{4}-\d{2}-\d{2}$/,
+    combine: dismantledDate => `${dismantledDate.year}-${dismantledDate.month}-${dismantledDate.day}`,
+    dismantle: (string) => {
+      const pieces = string.split('-');
+
+      return {
+        day: pieces[2],
+        month: pieces[1],
+        year: pieces[0],
+      };
+    },
+  },
+  'Month dd, yyyy': {
+    matcher: /^[ADFJMNOS][a-z]+ \d{2}, \d{4}$/,
+    combine: (dismantledDate) => {
+      const monthIndex = parseInt(dismantledDate.month, 10) - 1;
+      const month = months[monthIndex].charAt(0).toUpperCase() + months[monthIndex].slice(1);
+
+      return `${month} ${dismantledDate.day}, ${dismantledDate.year}`;
+    },
+    dismantle: (string) => {
+      const pieces = string.replace(',', '').split(' ');
+      const month = (months.indexOf(pieces[0].toLowerCase()) + 1).toString(10);
+
+      return {
+        day: pieces[1].padStart(2, '0'),
+        month: month.padStart(2, '0'),
+        year: pieces[2],
+      };
+    },
+  },
+};
+
+export function formatMatching(dateString) {
```

Comment:
We might not need to export this. I can't see a usage of it.

Reply:
I'm exporting it for testing purposes. Should I remove those specs?

Reply:
Yeah, we can remove the tests if there is sufficient test coverage on usages of it -- no spying!